geometry/eolearn/geometry/transformations.py

```diff
@@ -141,73 +143,69 @@ def _vector_is_timeless(vector_input):
+        """ Provides a vector dataframe either from the attribute or from given EOPatch feature
         """
         if self._is_geopandas_object(self.vector_input):
             return self.vector_input
 
-        feature_type, feature_name = next(self.vector_input(eopatch))
-        return eopatch[feature_type][feature_name]
+        feature = next(self.vector_input(eopatch))
+        return eopatch[feature]
 
-    def _sort_vector_timestamps(self, eopatch):
-        """ indexes all timestamps found in the vector input data according to the eopatch timestamps. Any
-        timestamps present in the vector data but missing from the EOPatch are ignored. The exception to this
-        is if no timestamps are found in the EOPatch, they are added from the vector data.
-        """
-        vectors = self._get_vector_data(eopatch)
-        timestamps64 = np.sort(vectors.TIMESTAMP.unique())
-        vector_timestamps = timestamps64.astype('M8[ms]').astype('O').tolist()
-
-        if not eopatch.timestamp:
-            eopatch.timestamp = list(vector_timestamps)
-            warnings.warn('The EOPatch has no timestamp data. Timestamps from the vector data will be added '
-                          'to the EOPatch', RuntimeWarning)
-
-        return vector_timestamps
-
-    def _get_rasterization_shapes(self, eopatch, group_classes=False, timeless=True):
-        """ Returns a generator of pairs of geometrical shapes and values. In rasterization process each shape will be
-        rasterized to it's corresponding value.
-        If there are no such geometries it will return `None`
-        :param group_classes: If true, function returns a zip that iterates through cascaded unions of polygons of the
-        same class, otherwise zip iterates through all polygons regardless of their class.
-        :type group_classes: boolean
-        :param timestamp: If supplied, only shapes associated with the given timestamp will be returned.
-        :type timestamp: datetime.datetime
+    def _preprocess_vector_data(self, vector_data, bbox, timestamps):
+        """ Applies preprocessing steps on a dataframe with geometries and potential values and timestamps
         """
+        columns_to_keep = ['geometry']
+        if self._rasterize_per_timestamp:
+            columns_to_keep.append('TIMESTAMP')
+        if self.values_column is not None:
+            columns_to_keep.append(self.values_column)
+        vector_data = vector_data[columns_to_keep]
 
-        vector_data = self._get_vector_data(eopatch)
+        if self._rasterize_per_timestamp:
+            vector_data = vector_data[vector_data.TIMESTAMP.isin(timestamps)]
 
-        if not timeless:
-            #remove timestamps that are not in the eopatch
-            vector_data = vector_data[vector_data['TIMESTAMP'].isin(eopatch.timestamp)]
+        if self.values_column is not None and self.values is not None:
+            values = [self.values] if isinstance(self.values, (int, float)) else self.values
```

Comment:
Why is this needed?

Reply:
This `self.values` can either be a single value (i.e. int/float) or a list of values. Perhaps this can already be parsed in init?

Reply:
Ah, my bad, i thought it was `values_column` twice, so it was odd that we were checking if its an integer or float.